### PR TITLE
Fix calibrations

### PR DESCRIPTION
**Pull request by jaib1**
Fixes for #4, #128, #164

If this type of fix looks good, let me know. I also want to improve the doc for hw.ptb.Window, make function/method names clearer, and create simple config scripts explaining how to run gamma calibration and reward valve calibration (these may share some redundancy with what goes on ReadTheDocs, but I think that's fine)

### CODE REVIEW

**Comment by k1o0** [comment]
Seems the tests are failing catastrophically.  How do they run on your rig?

File: +hw/+ptb/Window.m [hunk 1]
```diff
@@ -385,6 +386,28 @@ function fillRect(obj, colour, rect)
     end
 
     function [nx, ny] = drawText(obj, text, x, y, colour, vSpacing, wrapAt)
+    % Calls PTB's `DrawFormattedText` to display text to the PTB Window
+    %
+    % Inputs:
+    %   `text`: the text to be displaye
```
Comment (k1o0) [inline +hw/+ptb/Window.m]: typo

File: +srv/expServer.m [hunk 2]
```diff
@@ -14,17 +14,18 @@ function expServer(useTimelineOverride, bgColour)
 %       harware file is used.
 %
 %   Key bindings:
-%     t - Toggle Timeline on and off.  The default state is defined in the
+%     't' - Toggle Timeline on and off.  The default state is defined in the
```
Comment (k1o0) [inline +srv/expServer.m]: You're not being consistent with the inverted commas.  I find this just adds clutter, especially as they're followed by a hyphen anyway.  But if you're going to use them everywhere please stick to one type of comma.

File: +srv/expServer.m [hunk 3]
```diff
@@ -358,19 +359,30 @@ function whiteScreen()
 
   function calibrateGamma()
     stimWindow = rig.stimWindow;
+    % move PTB window to background and bring MATLAB command window to front
+    Screen('Preference', 'WindowShieldingLevel', 0);
+    stimWindow.close();
```
Comment (k1o0) [inline +srv/expServer.m]: Why close and open stimWindow?

File: +hw/+ptb/Window.m [hunk 4]
```diff
@@ -385,6 +386,28 @@ function fillRect(obj, colour, rect)
...
+    %   `colour`: an [r g b] or [r g b a] numeric vector for the color of 
+    %   the text
+    %   `vSpacing`: a number for the spacing between lines of text
+    %   `wrapAt`: a number for the max number of characters in a line of text
+    %
+    % Outputs:
+    %   `nx`: 
+    %   `ny`:
+    %
+    % Examples:
```
Comment (k1o0) [inline +hw/+ptb/Window.m]: Please remove this or add a TODO (or add examples)

File: +hw/+ptb/Window.m [hunk 5]
```diff
@@ -385,6 +386,28 @@ function fillRect(obj, colour, rect)
...
+    %   the text
+    %   `y`: a number, in pixels, of the y-position of the top border of 
+    %   the text, in pixels
+    %   `colour`: an [r g b] or [r g b a] numeric vector for the color of 
+    %   the text
+    %   `vSpacing`: a number for the spacing between lines of text
+    %   `wrapAt`: a number for the max number of characters in a line of text
+    %
+    % Outputs:
+    %   `nx`:
```
Comment (k1o0) [inline +hw/+ptb/Window.m]: May as well fill this in.  It's in the PTB docs

File: +hw/+ptb/Window.m [hunk 6]
```diff
@@ -385,6 +386,28 @@ function fillRect(obj, colour, rect)
     end
 
     function [nx, ny] = drawText(obj, text, x, y, colour, vSpacing, wrapAt)
+    % Calls PTB's `DrawFormattedText` to display text to the PTB Window
+    %
+    % Inputs:
+    %   `text`: the text to be displaye
+    %   `x`: a number, in pixels, of the x-position of the left border of
```
Comment (k1o0) [inline +hw/+ptb/Window.m]: Your formatting is really inconsistent.  Try to stick to what's already in the repository code instead of coming up with a new style each time.  you switch between `` ` `` and `'`, `:` and `-`.

File: +hw/+ptb/Window.m [hunk 7]
```diff
@@ -197,9 +197,10 @@ function open(obj)
       end
       Screen('Preference', 'SuppressAllWarnings', true);
       % setup screen window
+      %obj.PxDepth = 32;
```
Comment (k1o0) [inline +hw/+ptb/Window.m]: Instead of commenting things out with no explanation, leave some sort of description or remove entirely.  Otherwise it's just clutter that will mean nothing to no one in a few month's time.  

[MATLAB doesn't save parameters whose defaults are unchanged](https://uk.mathworks.com/help/matlab/matlab_oop/understanding-the-save-and-load-process.html).  This means that you might end up breaking someone's rig by making this change.  If you want to go ahead with it let everyone know on slack to check this works on their rigs _before_ merging into master.

File: +srv/expServer.m (hunk 3 above)
Comment (jaib1) [reply +srv/expServer.m]: currently open PTB windows are not affected when `'WindowShieldingLevel'` changes

File: +hw/+ptb/Window.m (hunk 4 above)
Comment (jaib1) [reply +hw/+ptb/Window.m]: yup, this goes along with my comment in the initial PR where I say:

> I also want to improve the doc for hw.ptb.Window
_if_ you approve of this PR's general approach to solving #4, #128, #164 (I didn't want to waste time on documentation if you didn't approve of the code approach =D)

File: +hw/+ptb/Window.m (hunk 6 above)
Comment (jaib1) [reply +hw/+ptb/Window.m]: ok, I'll make the formatting consistent

File: +hw/+ptb/Window.m (hunk 7 above)
Comment (jaib1) [reply +hw/+ptb/Window.m]: this was a little egg for you : ) want to see if there's any reason it's set to 32? 
Regarding the not saving of this property, I tested it out on the training rigs and BII and all was good, so i think should be fine, but I can send a message to the slack as well

File: +hw/+ptb/Window.m (hunk 5 above)
Comment (jaib1) [reply +hw/+ptb/Window.m]: yup, this goes along with my comment in the initial PR where I say:

> I also want to improve the doc for hw.ptb.Window

_if_ you approve of this PR's general approach to solving #4, #128, #164 (I didn't want to waste time on documentation if you didn't approve of the code approach =D)

**Comment by jaib1** [comment]
Haven't run the current tests on this yet because I wanted your general opinion on this PR. I can check tomorrow. And yeah, I'll add test(s) here too.

File: +hw/+ptb/Window.m (hunk 4 above)
Comment (k1o0) [reply +hw/+ptb/Window.m]: Yeah looks good to me

Comment (k1o0) [reply +hw/+ptb/Window.m]: (Documenting existing code is never a waste of time! :)

File: +hw/+ptb/Window.m (hunk 7 above)
Comment (k1o0) [reply +hw/+ptb/Window.m]: I don't know enough about graphics tbh but there's often a good reason for things to be set as they are.  It may be that controlling the bit depth is essential for generating the visual stimuli correctly.  If you let the system determine the value I suppose it will vary across rigs which we don't necessarily want.  I know that bit depth is important in rendering low contrasts and that there are some assumptions made in parts of the code (e.g. https://github.com/cortex-lab/signals/blob/9f4f5d068805d8dc715e332449086e42fa396da2/%2Bvis/gaussianLayer.m#L18-L22).  From what I've read 32 is an odd value for bit depth, and it may simply be a carry-over from when we used a particular monitor.

Long story short, I've come to realize that one shouldn't flippantly change bits of code when you don't understand them.  You should ask someone like Krumin about this and make sure everyone tests their recording rigs or at least approves this change.

Comment (jaib1) [reply +hw/+ptb/Window.m]: yup fair point, I'll do this
